lib/services/android-debug-service.ts

```diff
@@ -186,7 +193,28 @@ class AndroidDebugService implements IDebugService {
 
 			this.device.applicationManager.stopApplication(packageName).wait();
 			this.device.applicationManager.startApplication(packageName).wait();
-
+			
+
+			let waitText: string = `0 /data/local/tmp/${packageName}-debugbreak`;
+			let maxWait = 6;
+			let debugerStarted: boolean = false;
+			while (maxWait > 0 && !debugerStarted) {
+				let forwardsResult = this.device.adb.executeShellCommand([`ls -s /data/local/tmp/${packageName}-debugbreak`]).wait();
+				maxWait--;
+				debugerStarted = forwardsResult.indexOf(waitText) === -1;
+				if (!debugerStarted) {
+					this.sleep(500);
+				}	
+			}
+			
+			if (debugerStarted) {
+				console.log("# NativeScript Debugger started #");
+			}
+			else {
+				console.log("# NativeScript Debugger did not start in time #");
```

Comment:
We use `$logger` so that we can write unit tests, please use it to aid us when we later add tests.

If the text printed with `console.log` is designed to be consumed by a tool (VS Code) and not a human user, I suggest to hide it behind a command-line flag so that it does not complicate the user's work flow.

Filtering out $logger in production code (not unit tests) would completely break the CLI and I do not see how could this happen.